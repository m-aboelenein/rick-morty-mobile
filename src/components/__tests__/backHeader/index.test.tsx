import {fireEvent, render} from '@testing-library/react-native';
import * as React from 'react';
import renderer from 'react-test-renderer';
import BackHeader from '../../backHeader';

const goBack = jest.fn();

it('renders a Back Header', () => {
  const tree = renderer.create(<BackHeader goBack={goBack} />);
  expect(tree).toMatchSnapshot();
});

it('navigates back to characters screen', () => {
  const {getByTestId} = render(<BackHeader goBack={goBack} />);

  fireEvent.press(getByTestId('back-button'));
  expect(goBack).toHaveBeenCalled();
});
