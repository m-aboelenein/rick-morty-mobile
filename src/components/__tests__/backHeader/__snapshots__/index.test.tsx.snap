// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Back Header 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flexDirection": "row",
      "height": 40,
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
    testID="back-button"
  >
    <svg
      data-file-name="SvgChevronLeft"
      height={25}
      width={25}
    />
  </View>
</View>
`;
